PySDM/backends/impl_thrust_rtc/methods/collisions_methods.py

```diff
@@ -252,141 +256,92 @@ def __init__(self):
             """,
         )
 
-        self.__feingold1988_fragmentation_body = trtc.For(
-            param_names=("scale", "frag_size", "x_plus_y", "rand", "fragtol"),
-            name_iter="i",
-            body="""
-            auto log_arg = max(1 - rand[i] * scale / x_plus_y[i], fragtol);
-            frag_size[i] = -scale * log(log_arg);
-            """,
-        )
+        if self.formulae.fragmentation_function.__name__ == "Feingold1988Frag":
+            self.__feingold1988_fragmentation_body = trtc.For(
+                param_names=("scale", "frag_size", "x_plus_y", "rand", "fragtol"),
+                name_iter="i",
+                body=f"""
+                frag_size[i] = {self.formulae.fragmentation_function.frag_size.c_inline(
```

Comment:
Referring to comment https://github.com/atmos-cloud-sim-uj/PySDM/pull/945/files#r1008950308

This could be smth like `Formulae.get_c_inline(Feingold1988Frag.frag_size)`.  And we will not need the `if` statement 5 lines higher 
https://github.com/atmos-cloud-sim-uj/PySDM/blob/c035cba2ab866e56ba68c77f6a7cf46cb9fafb4b/PySDM/backends/impl_thrust_rtc/methods/collisions_methods.py#L259

Reply:
The `c_inline` would at least need to get the constants dict somehow (the first arg to all "physics" functions).

Reply:
if do this using `self.formulae` instance e.g. `self.formulae.c_inline(Feingold1988Frag.get_frag_size)`, will it work?

Reply:
in principle, we could make it work, but the goal of `Formulae` is to pick from variants of its components during instantiation of `Formulae` so that when calling it we just refer to a component, and not to a variant of it. So rather:
```
self.formulae.c_inline('fragmentation_function', 'get_frag_size')
```
?